torch/csrc/autograd/function.h

```diff
@@ -97,6 +108,13 @@ struct TORCH_API Node : std::enable_shared_from_this<Node> {
       next_edges_(std::move(next_edges)) {
     if (AnomalyMode::is_enabled()) {
       metadata()->store_stack();
+
+    }
+    // If anomaly mode is enabled and graph is constructed, then assign this
+    // node to a parent. A parent is a Node where this Node is created.
+    // We are tracking the parents to track multiple backward operations.
+    if (GradMode::is_enabled()) {
```

Comment:
You don't want to always do that, only in anomaly mode. You can move it inside the if statement above.
This is a critical code path and we don't want to do extra work unless it is absolutely necessary.

Reply:
Ah, sorry! This is bug from my side. What I wrote as the comment is different from what the code does.